Backbone.API/DevicesDbContextSeed.cs

```diff
@@ -75,20 +81,16 @@ private async Task SeedBasicTier(DevicesDbContext context)
 
     private async Task AddBasicTierToIdentities(DevicesDbContext context)
     {
-        try
-        {
-            var basicTier = await context.Tiers.BasicTier(CancellationToken.None);
-            var identitiesWithoutTier = from i in context.Identities where i.Tier == null select i;
-            await identitiesWithoutTier.ForEachAsync(it =>
-            {
-                it.Tier = basicTier;
-            });
-
-            await context.SaveChangesAsync();
-        }
-        catch (Exception)
+        var basicTier = await context.Tiers.GetBasicTier(CancellationToken.None);
+        if(basicTier == null)
+            return;
+
+        var identitiesWithoutTier = context.Identities.Where(i => i.Tier == null).ToList();
+        identitiesWithoutTier.ForEach(it =>
         {
-            throw new Exception("Basic Tier was not found.");
-        }
+            it.Tier = basicTier;
+        });
+
+        await context.SaveChangesAsync();
```

Comment:
You should replace it with

```c#
await context.Identities.Where(i => i.Tier == null).ExecuteUpdateAsync(s => s.SetProperty(i => i.Tier, basicTier));
```
in order to reduce the number of database roundtrips. Depending on the number of Identities, this could make a huge difference.

I never used the `ExecuteUpdateAsync` method before, so I don't know if it works with navigation properties.

Reply:
I didn't pay a lot of attention to this because I supposed all changes would be commited at once on `await context.SaveChangesAsync();`.

It seems EF cannot handle it when we set a new navigation property. I'm trying to workaround the issue.

Reply:
> I supposed all changes would be commited at once on `await context.SaveChangesAsync();`.

They are commited at once. But you still need 2 roundtrips: one for the update and one for the select (line 88).

If my suggested solution doesn't work for navigation properties, you should at least call `ToListAsync` instead of `ToList` in line 88.